metrics/src/vespa/metrics/metricmanager.cpp

```diff
@@ -32,7 +32,7 @@ MetricManager::Timer::getTime() const {
 
 void
 MetricManager::assertMetricLockLocked(const MetricLockGuard& g) const {
-    if ((g.mutex() != &_waiter) || !g.owns_lock()) {
+    if (g.owns(_waiter)) {
```

Comment:
Should this be `!g.owns(_waiter)` instead ?

Reply:
Agreed, this looks like an error in the PR.

Reply:
Good catch. The effect of -1 * -1 = 1.
Fixed